vsphere/resource_vsphere_resource_pool.go

```diff
@@ -0,0 +1,255 @@
+package vsphere
+
+import (
+	//"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-vsphere/vsphere/internal/helper/resourcepool"
+	"github.com/terraform-providers/terraform-provider-vsphere/vsphere/internal/helper/structure"
+	"github.com/terraform-providers/terraform-provider-vsphere/vsphere/internal/virtualdevice"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+const resourceVSphereResourcePoolName = "vsphere_resource_pool"
+
+func resourceVSphereResourcePool() *schema.Resource {
+	s := map[string]*schema.Schema{
+		"name": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "Name of resource pool.",
+		},
+		"root_resource_pool_id": {
```

Comment:
This should be changed to `parent_resource_pool_id`, as the current language implies that this field can only be used to supply _root_ resource pools, where in reality you can create resource pools for as many layers of recursion as you need (with each child pool slicing off portions of its parent). Documentation language should be modified as needed as well.